File: backend/domains/task/repository/interface.go
package repository

import (
	"context"

	"github.com/erweixin/go-genai-stack/backend/domains/task/model"
)

// TaskRepository 定义任务仓储接口
//
// 遵循领域驱动设计原则，提供对 Task 聚合根的持久化操作
type TaskRepository interface {
	// Create 保存一个新的任务
	Create(ctx context.Context, task *model.Task) error

	// FindByID 根据 ID 查找任务
	FindByID(ctx context.Context, taskID string) (*model.Task, error)

	// Update 更新一个现有任务
	Update(ctx context.Context, task *model.Task) error

	// Delete 根据 ID 删除任务
	Delete(ctx context.Context, taskID string) error

	// List 根据筛选条件列出任务
	// 返回任务列表、总数和错误
	List(ctx context.Context, filter *TaskFilter) ([]*model.Task, int, error)

	// Exists 检查任务是否存在
	Exists(ctx context.Context, taskID string) (bool, error)
}

File: backend-nodejs/domains/task/repository/interface.ts
/**
 * Task Repository 接口
 * 定义任务仓储的抽象接口
 */

import type { Task } from '../model/task.js';

export interface TaskFilter {
  // 筛选条件
  userId?: string;
  status?: 'pending' | 'in_progress' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  tag?: string;
  dueDateFrom?: Date;
  dueDateTo?: Date;
  keyword?: string;

  // 排序
  sortBy?: 'created_at' | 'due_date' | 'priority';
  sortOrder?: 'asc' | 'desc';

  // 分页
  page: number;
  limit: number;
}

/**
 * TaskRepository 任务仓储接口
 */
export interface TaskRepository {
  /**
   * 创建任务
   */
  create(ctx: unknown, task: Task): Promise<void>;

  /**
   * 根据 ID 查找任务
   */
  findById(ctx: unknown, taskId: string): Promise<Task | null>;

  /**
   * 更新任务
   */
  update(ctx: unknown, task: Task): Promise<void>;

  /**
   * 删除任务
   */
  delete(ctx: unknown, taskId: string): Promise<void>;

  /**
   * 列出任务
   */
  list(ctx: unknown, filter: TaskFilter): Promise<{ tasks: Task[]; total: number }>;

  /**
   * 检查任务是否存在
   */
  exists(ctx: unknown, taskId: string): Promise<boolean>;
}

